README.md

```diff
@@ -73,6 +73,25 @@ Whenever the same functionality can be achieved in multiple offerings but in a d
 
 For links within this documentation, please end the link with `/index.md`. This way all links will be checked, and you’ll get an error if a link is broken at any point.
 
+### Concepts
+
+There are several key concepts at Snowplow: events (self-describing, structured), entities, schemas. We must ensure that we use and explain them consistently.
+
+**Please, use up-to-date terms:**
+* “Self-describing event”, not “unstructured event”
+* “Entities”, not “contexts”
+* If you are writing about schemas, pick “schema” or “data structure” and stick with it
+
```

Comment:
Can we also agree how to refer to the entity that is the collection of schema-defined-objects, and the entity that is an individual object defined by the schema?

Reply:
entity (single object) > entities (list of object) > entitieses (multiple different lists of objects) 🙃

Reply:
Hmmmmm, is that a thing (the list of lists)? Where is it found?

Reply:
entity, entities, sets of entities? But where does the need for such terms arise?

Reply:
I'd say we actually use entity, entity, then entities currently. Because we say the YAUAA entity, in e-commerce we use the product entity (but it can be multiple sent at the same time), but you can attach multiple entities to the same event... Feels like we don't have a way to distinguish between the object itself and the list of objects that end up in the same row+column

Reply:
This was why I was considering context for the list of entities, but I think it's because I'm just used to working with the columns named `context_...`

Reply:
I do say that in the page about entities:
> especially to refer to a set of _multiple_ entities

However, I am not fond of the idea of making this meaning too official, for two reasons:
* It can create more confusion, because one thing is to move from “context” to “entity” and from “contexts” to “entities” (easy to explain), and another is to move from “context” to “entity” and from “contexts” to “context” (???)
* “Context” is a rather generic and common word, so making it a part of a controlled vocabulary feels hopeless. There will inevitably be sentences like “blah blah this provides some context to the event”. I would prefer these sentences to casually make sense (as they do), rather than mean a very specific thing (especially when this specific thing brings the confusion I mention in the previous point).

Not sure if these are rational arguments but it’s my best attempt to explain why I don’t like this :D
*

Reply:
I agree we shouldn't really be mixing terms at that level - I think ideally you want the set of lists to be a plural version of list of objects, but then you're left wondering what to call the individual object. We need plural plurals in English is clearly what's causing problems.

You almost want entity<context (list of entities)<contexts (multiple contexts), but I know you don't like that. I think the inverse of that is worse because a single object being a context feels linguistically wrong...

Reply:
I’ve been using:
* “entity” to mean a specific entity object
* “entities” to mean a set of entity objects, of the same type or different types
* “types of entities” to mean a set of distinct entity schemas, where it matters to make this distinction (e.g.: “Each type of entity gets its own column”)

Notice that we have the same exact problem with events. We use the word “event” both to mean a particular event and also to mean “type of event” (a “blueprint” for a particular event).

Reply:
I am still unsure where “multiple contexts” would be useful beyond the situation where we want to say that event table has multiple columns for each type of entity. (Btw — I just said it and did not need to use the plural of plural!)

Reply:
But there you are using the singular of entity to describe the list version, which may itself be multiple entity objects! We can't (I mean, we do, but we shouldn't) have the exact same word refer to both an individual instance of a thing (defined by the schema) and the list of that thing which is attached to an event.

Reply:
But you don't want to. say "...has multiple columns, one for each type of entities" because that's way worse...